src/connector/src/source/cdc/external/mysql.rs

```diff
@@ -543,9 +609,30 @@ impl MySqlExternalTableReader {
                         let ty = field_map.get(pk.as_str()).unwrap();
                         let val = match ty {
                             DataType::Boolean => Value::from(value.into_bool()),
-                            DataType::Int16 => Value::from(value.into_int16()),
-                            DataType::Int32 => Value::from(value.into_int32()),
-                            DataType::Int64 => Value::from(value.into_int64()),
+                            DataType::Int16 => {
+                                let int16_val = value.into_int16();
+                                if int16_val < 0 && self.is_unsigned_type(pk.as_str()) {
```

Comment:
What if a later refactoring removes `int16_val < 0`? Will existing test cases catch the error, or it is safe to remove?

Reply:
It's safe to remove those `int16_val < 0` judgement, because the subsequent check of `self.is_unsigned_type` will be successful. This is added here to reduce overhead caused by checking `is_unsigned_type`, based on the principle of prioritizing the left-hand side.